EKrut-Client/src/ekrut/client/managers/ClientTicketManager.java

```diff
@@ -0,0 +1,86 @@
+package ekrut.client.managers;
+
+import java.util.ArrayList;
+
+import ekrut.entity.Ticket;
+import ekrut.net.ResultType;
+import ekrut.net.TicketRequest;
+import ekrut.net.TicketRequestType;
+import ekrut.net.TicketResponse;
+import ekrut.net.UserRequest;
+import ekrut.net.UserResponse;
+
+//need to add- sendRequest
+
+
+/**
+ * The ClientTicketManager class is responsible for managing tickets for a client.
+ * This includes updating the status of a ticket and fetching tickets.
+ * 
+ * @author Noy Malka
+ */
+
+public class ClientTicketManager {
+
+	/**
+     * Updates the status of a given ticket.
+     * 
+     * @param ticket the ticket to update
+     * @return the result of the update operation
+     * @throws NullPointerException if ticket is null
+     * @throws RuntimeException if there is an error updating the ticket's status
+     */
+	
+	public ResultType updateTicketStatus(Ticket ticket) throws Exception{
+		if (ticket==null) {
+			throw new NullPointerException("provided null ticket");

```

Comment:
Prefer to throw `IllegalArgumentException` instead. Throwing `NullPointerException` can be problematic for various reasons.